projects/packages/waf/src/class-waf-cli.php

```diff
@@ -65,15 +65,92 @@ public function mode( $args ) {
 				sprintf(
 					/* translators: %1$s is the name of the mode that was just switched to. */
 					__( 'Jetpack WAF mode switched to "%1$s".', 'jetpack-waf' ),
-					get_option( WafRunner::MODE_OPTION_NAME )
+					get_option( Waf_Runner::MODE_OPTION_NAME )
 				)
 			);
 		}
 		WP_CLI::line(
 			sprintf(
 				/* translators: %1$s is the name of the mode that the waf is currently running in. */
 				__( 'Jetpack WAF is running in "%1$s" mode.', 'jetpack-waf' ),
-				get_option( WafRunner::MODE_OPTION_NAME )
+				get_option( Waf_Runner::MODE_OPTION_NAME )
+			)
+		);
+	}
+
+	/**
+	 * Setup the WAF to run.
+	 * ## OPTIONS
+	 *
+	 * [<mode>]
+	 * : The new mode to be set.
+	 * ---
+	 * options:
+	 *   - silent
+	 *   - normal
+	 * ---
+	 *
+	 * @param array $args Arguments passed to CLI.
+	 * @return void|null
+	 * @throws WP_CLI\ExitException If there is an error switching the mode.
+	 */
+	public function setup( $args ) {
+		if ( ! Waf_Runner::is_allowed_mode( $args[0] ) ) {
+
+			return WP_CLI::error(
+				sprintf(
+					/* translators: %1$s is the mode that was actually found. Also note that the expected "silent" and "normal" are hard-coded strings and must therefore stay the same in any translation. */
+					__( 'Invalid mode: %1$s. Expected "silent" or "normal".', 'jetpack-waf' ),
+					$args[0]
+				)
+			);
+		}
```

Comment:
I played around with that a bit, but came to a similar conclusion that the benefit of abstracting it out is not worth the awkwardness that doing so creates. In the end it feels like just adding an unnecessary wrapper around is_allowed_mode. If we end up with more CLI commands and find that the same code shows up more then it may be worth looking at creating a shortcut of sorts to that error - like `return invalid_mode_error();` or something.